### PR TITLE
Refactor Sequencer

### PR DESCRIPTION
**Pull request by sieren**
ping @analogcode @marcussatellite 

This is my first iteration of the Sequencer Refactor.

In the first commit I'm extracting the sequencer code from the Kernel into a separate class.
The second commit cleans up and extracts the Arpegiator code into separate static functions.
I also consolidate the Down and DownMinusTail ones into one function.

Tested on my end, but I'd feel way more comfortable if you all would give this a spin too before merging :)

### CODE REVIEW

**Comment by sieren** [comment]
ping @aure maybe you can take a look too since you were involved in the first refactoring :)

File: AudioKitSynthOne/DSP/Sequencer/S1Sequencer.mm [hunk 1]
```diff
@@ -0,0 +1,198 @@
+//
+//  S1NoteState.mm
```
Comment (aure) [inline AudioKitSynthOne/DSP/Sequencer/S1Sequencer.mm]: Wrong header, copy paste error?

File: AudioKitSynthOne/DSP/Sequencer/S1Arpegiator.hpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
+//
+//  S1Arpegiator.hpp
```
Comment (aure) [inline AudioKitSynthOne/DSP/Sequencer/S1Arpegiator.hpp]: Arpeggiator has two 'g's

**Comment by sieren** [comment]
@aure fixed those

**Comment by aure** [comment]
> @aure fixed those

Couple more, search for Arpegiator, like in ArpegiatorMode.  Or else just merge and I'll deal with trivialities.

**Comment by marcussatellite** [comment]
@aure Please don't merge yet...I'm working with Matthias on a potential bug in "up down" mode

**Comment by sieren** [comment]
@marcussatellite please take a look at the fixup! Let me know if it works for you and i'll squash/merge that one :)

**Pull request merged**